### ISSUE TITLE
'beforeAll' hook on Pest.php is executed more than once for a single file

### ISSUE THREAD

**Issue by fnzr** [pestphp/pest#351]
Hi,
I'm trying to register a global hook that will be executed once before all tests, but when registering it on the Pest.php file, it is called once per test instead.
Reproducible from this repository: https://github.com/fnzr/pest-sample-temp
What I've done:
1) `composer require pestphp/pest --dev --with-all-dependencies`
2) `./vendor/bin/pest --init`
3) Created the sample tests:
```php 
//Pest.php
uses()->beforeAll(function() {
    echo 'Global: beforeAll' . PHP_EOL;
})->in('subdir');
```
```php
//subdir/MyTest.php
beforeAll(function(){
    echo 'Subdir: beforeAll'  . PHP_EOL;
});

test('first', function() {
    $this->assertTrue(true);
});

test('second', function() {
    $this->assertTrue(true);
});
```
3) `./vendor/bin/pest`

Result:
```
Global: beforeAll
Global: beforeAll
Subdir: beforeAll

   PASS  Tests\subdir\MyTest
  ✓ first
  ✓ second

  Tests:  2 passed
  Time:   0.05s
```
Adding a third test,  "Global: beforeAll" is shown 3 times. Note that "Subdir: beforeAll" shows up only once.
I expected "Global: beforeAll" to appear only once. Did I misunderstand something?

Edit: ah, and I'm on PHP 8.0.8

**Comment by jordanbrauer** [comment]
Interesting. Apologies if this has been a blocker for you! 

I'll take a look at this and see if we can't get it reproduced with a test case to start on a fix. 🙂

**Comment by fnzr** [comment]
I've been debugging it a bit, and all repeated calls to the global setup on Pest.php are called before any test runs.

It seems that for each test in the file, the global beforeAll hook is being stacked on the Testable::$beforeAll variable.  When Testable::setUpBeforeClass executes the $beforeAll closure on line 198, the entire Testable::$beforeAll stack is executed (the exact function call to the global hook happens on the closure returned by ChainableClosure::fromStatic, line 32). 

Basically, the problem happens because each test, when it's being parsed, registers the global hook on a shared variable (Testable::$beforeAll) as if it were a new hook. The solution would be...well, not do that.  Sorry, I dont understand the library enough to propose a fix.

**Comment by jordanbrauer** [comment]
@fnzr no worries, thank you for the information! Should be useful when whipping a fix up. Sounds like we need to do some sort of reset in this case.

**Comment by jordanbrauer** [comment]
Just a small update @fnzr!

I've got a fix and will send a PR in for review tonight. Reason for delay was the test that backs it is a bit finicky due to the unpredictable nature of how many times these get called – _i.e._, running a specific test, some tests, or all tests yield a different number of calls to these hooks.
